### ISSUE TITLE
Hot reloading doesn't work

### ISSUE THREAD

**Issue by entropitor** [archriss/react-native-snap-carousel#166]
It crashes with the error `Cannot read property 'scrollToIndex' of null` and a warning of `Can only update a mounted or mounting component. This usually means you called setState, replaceState or forceUpdate on a unmounted component. This is a no-op. Please check the code for the Carousel component`

**Comment by entropitor** [comment]
I'm guessing componentWillReceiveProps is called again which call snapToItem which calls scrollToIndex on a null reference to `this._flatlist`

**Comment by jerolimov** [comment]
Hey, as written in your PR #167, we have this in our app as well. In our case, this has nothing odo with the hot reloading. Its happen in our production app / build.

Will try your PR in the next app update.

```
null is not an object (evaluating 'this._listRef.scrollToOffset')

Stack:
- value@index.android.bundle:205:2194
- index.android.bundle:994:5820
- index.android.bundle:23:1366
- o@index.android.bundle:23:571
- callTimers@index.android.bundle:23:2778
- value@index.android.bundle:18:2975
- index.android.bundle:18:280
- value@index.android.bundle:18:2426
- callFunctionReturnFlushedQueue@index.android.bundle:18:252
```

**Comment by jonching** [comment]
I also get this error (Cannot read property 'scrollToIndex' of null) in my app. It happens whenever I remove the snap carousel from the DOM of a view that it was previously showing in.

**Comment by bd-arc** [comment]
As mentioned in the PR discussion, the issue should be fixed in the "soon-to-be-merged" [`loop` branch](https://github.com/archriss/react-native-snap-carousel/tree/loop). I would definitely appreciate your confirmation ;-)

To try out the new branch:
```
"react-native-snap-carousel": "https://github.com/archriss/react-native-snap-carousel#18acafa"

**Comment by entropitor** [comment]
It still crashes

**Comment by entropitor** [comment]
It seems to work

**Comment by bd-arc** [comment]
Fix released in version `3.3.0`.
